src/browser-sandbox/pageLoadSkipTimeout.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Sandbox function to stop requests after page load timeout
+ *
+ * @param pageLoadSkipTimeout
+ * @returns {*}
+ */
+
+export default function pruneNonCriticalSelectors ({ pageLoadSkipTimeout }) {
+  console.log('debug: pageLoadSkipTimeout [' + pageLoadSkipTimeout + ']')
+  try {
+    return new Promise((resolve, reject) => {
+      if (pageLoadSkipTimeout) {
+        const start = Date.now()
+        const pageLoadTimeout = function () {
+          window.requestAnimationFrame(() => {
+            const timePassed = Date.now() - start
+
+            if (timePassed >= pageLoadSkipTimeout) {
+              window.stop()
```

Comment:
The intention with the Promise.race for the page load in `master` is just to prevent Penthouse to take too long, when the page has reasonably loaded, but due to imperfect detection mechanisms puppeteer still believes it has not.

Reply:
The mindset of stop waiting for the page to load but let it load while extracting the css is helping me to understand. 
As mentioned above I guess it is triggering the "load" event which results in a resolving of the promise. Which is what I want to achieve.